### PR TITLE
`az aro update` `--load-balancer-managed-outbound-ip-count` flag support

### PR DESCRIPTION
**Pull request by tsatam**
### Which issue this PR addresses:

Fixes [ARO-3103](https://issues.redhat.com/browse/ARO-3103)
Implicitly fixes [ARO-2467](https://issues.redhat.com/browse/ARO-2467) as well

### What this PR does / why we need it:

Implements support within the `az aro update` command in the ARO extension for the managedOutboundIps.Count property on the cluster load balancer profile, implemented in the 2023-07-01 preview API. 

### Test plan for issue:

On a provisioned ARO cluster, managed by an RP with the 2023-07-01 preview API (e.g. local dev):

- Run `az aro update --help` and validate that the `--load-balancer-managed-outbound-ip-count` flag and information is present
- Run `az aro update --load-balancer-managed-outbound-ip-count $NUMBER` for numbers != 1 and ensure that the resulting cluster document has property `openShiftCluster.properties.networkProfile.loadBalancerProfile.managedOutboundIps.count` set to `$NUMBER` instead of the previous value. 

### Is there any documentation that needs to be updated for this PR?

None for this specific PR (overall doc updates captured in [ARO-3513](https://issues.redhat.com/browse/ARO-3513))

### Additional Notes:

- Azure CLI command flags require at least one option to be present that is overall shorter than 22 characters. As a result, I have added the `--lb-ip-count` alias for this flag. Open to suggestions for name changes to this shorthand flag (as long as it is within the 22 character limit)
- This command will not yet perform any observable changes in ARO clusters, outside of changing the property within the cluster document, as [ARO-3110](https://issues.redhat.com/browse/ARO-3110) needs to be completed in order to implement the functionality in the RP.

### CODE REVIEW

**Comment by SudoBrendan** [comment]
Adding `hold` until we get BU approval on the shorthand.

File: python/az/aro/azext_aro/_params.py
Comment (SudoBrendan) [inline python/az/aro/azext_aro/_params.py:128]: Can we align this closer to the swagger spec? https://github.com/Azure/ARO-RP/blob/master/swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2023-07-01-preview/redhatopenshift.json#L2009

File: python/az/aro/azext_aro/_validators.py
Comment (SudoBrendan) [inline python/az/aro/azext_aro/_validators.py:283]: probably worth adding `(inclusive)` at the end of this message.

**Review by SudoBrendan — COMMENTED** [review]
Just a few comments on customer experience things, might be considered nits, but it would be good to align our messaging with the API.

**Review by SudoBrendan — COMMENTED** [review]
LGTM, but waiting on approval until I talk with the BU. Great work, thank you!

**Comment by tsatam** [comment]
NOTE: For developer convenience reasons, this PR should also be on hold until the 20230701preview API is released to all RP regions in prod, as it not being available will break local extension usage.

File: python/az/aro/azext_aro/_params.py
Comment (tsatam) [reply python/az/aro/azext_aro/_params.py:130]: I saw the shorthand alternative rule as a linter failure from the `azdev` linter commands. It's possible that this rule is ignorable, or that AKS implemented this flag before that rule was imposed and the rule only applies to new and/or changed flags. In any case I'm not sure how we could validate the rule being required until we attempt to upstream this change. 

+1 on aligning the description with AKS, I will make that change.

File: python/az/aro/azext_aro/_params.py
Comment (jhoreman) [inline python/az/aro/azext_aro/_params.py:118]: Comment: Upstream specifically wanted int32 formats here. Should work just fine but just calling it out.

Comment (SudoBrendan) [reply python/az/aro/azext_aro/_params.py:118]: I think that Tanmay had mentioned that we should likely hold on this until we do the downstreaming of all the REST Specs work, then rebase this PR on that.

Comment (tsatam) [reply python/az/aro/azext_aro/_params.py:118]: A quick peek at AKS's equivalent flag shows they've defined the type as `int` as well: https://github.com/Azure/azure-cli/blob/d419ab357481e803d2f2a84a40760fa6e6af7c8b/src/azure-cli/azure/cli/command_modules/acs/_params.py#L334

I'm not sure if the API specificity on the data type necessarily correlates to the CLI/Python-side requirements here.

Comment (tony-schndr) [reply python/az/aro/azext_aro/_params.py:118]: native python3 doesn't have a int32 type

**Comment by tsatam** [comment]
Updated PR to point to new `azext-aro-v20230701preview` branch, in order to not conflict with the in-flight efforts on the stable v20230901 API and corresponding CLI changes.

**Review by SudoBrendan — APPROVED** [review]

File: python/az/aro/azext_aro/custom.py
Comment (tony-schndr) [inline python/az/aro/azext_aro/custom.py:157]: Missed during rebase I'm guessing

File: python/az/aro/azext_aro/_params.py
Comment (SudoBrendan) [reply python/az/aro/azext_aro/_params.py:120]: are we good to resolve this?

**Review by SudoBrendan — CHANGES_REQUESTED** [review]
Other than the comment Tony left, LGTM.

File: python/az/aro/azext_aro/_params.py
Comment (SudoBrendan) [reply python/az/aro/azext_aro/_params.py]: This looks resolved as well, can we resolve?

File: python/az/aro/azext_aro/custom.py
Comment (tsatam) [reply python/az/aro/azext_aro/custom.py]: fixed, thanks.

File: python/az/aro/azext_aro/_params.py
Comment (tsatam) [reply python/az/aro/azext_aro/_params.py:130]: The description is not fixed, but we will be moving that effort to a post-release activity as it will require us to change how the entire `az aro` command is documented (moving user-facing documentation from the parameter definitions themselves into `_help.py`).

**Review by tony-schndr — APPROVED** [review]
lgtm, thanks Tanmay!

**Review by SudoBrendan — APPROVED** [review]

**Pull request merged**
